### PR TITLE
core/local/steps/win_detect_move: Fix move from inside move

### PR DESCRIPTION
**Pull request by sebn**
**Do not review yet**

Please make sure the following boxes are checked:

- [ ] PR is not too big
- [ ] it improves UX & DX in some way
- [ ] it includes tests matching the implementation changes
- [ ] it includes relevant documentation

### CODE REVIEW

File: core/local/steps/win_detect_move.js [hunk 1]
```diff
@@ -21,108 +23,188 @@ const DELAY = 1000
...
+    release()
+  }
+}
+
+async function assignDebugInfos (event, deletedIno, oldPaths) {
+  if (event.action !== 'deleted') return
+  if (oldPaths) {
+    _.set(event, [STEP_NAME, 'oldPaths'], oldPaths)
+  } else if (!deletedIno) {
+    _.set(event, [STEP_NAME, 'docNotFound'], 'missing')
```
Comment (sebn) [inline core/local/steps/win_detect_move.js]: `{inoNotFound: true}`? `{missingIno: true}`? (will require updating unit tests)

File: core/local/steps/win_detect_move.js [hunk 2]
```diff
@@ -21,108 +23,188 @@ const DELAY = 1000
...
 
-function sendReadyBatches (waiting /*: PendingBatch[] */, out /*: Buffer */) {
+async function findDeletedIno (event, pouch, unmergedRenamedEvents) {
+  if (event.action !== 'deleted') return {}
+  // OPTIMIZE: Make .previousPaths() include event.path so we don't need a lock
+  const release = await pouch.lock('winMoveDetector')
+  try {
+    return (
+      await findDocById(event._id, pouch) ||
+      await findDocRecentlyRenamed(
```
[the remark this answers is not in the document]
Comment (taratatach) [reply core/local/steps/win_detect_move.js]: It could :)

File: core/utils/sorted_set.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+/* @flow */
+
+module.exports = class SortedSet/* ::<A> */ {
```
[the remark this answers is not in the document]
Comment (taratatach) [reply core/utils/sorted_set.js]: I created a Trello card to do this later.

File: core/utils/sorted_set.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  delete (value /*: A */) {
+    if (this._set.delete(value)) {
+      const index = this._values.indexOf(value)
+      this._values.splice(index, 1)
+      return true
+    }
+    return false
+  }
+
+  reduceRight/* ::<B> */ (callback /*: (B, A) => B */, initialValue /*: B */) /*: B */{
```
[the remark this answers is not in the document]
Comment (taratatach) [reply core/utils/sorted_set.js]: It is seen by Flow and I wanted to have one way of writing it (in method calls we can't have spaces around the comment or eslint will complain and in this situation, not having a space after `/*` and before `*/` will generate another linter issue).

**Comment by sebn** [comment]
Merging as soon as CI is green.

**Pull request merged**